content/tutorial/tutorial.md

```diff
@@ -1042,20 +1049,21 @@ Let's `map` over the `history` in the Game's `render` method:
   }
 ```
 
-**[View the full code at this point](https://codepen.io/gaearon/pen/EmmGEa?editors=0010)**
+**[अभी तक का पूरा कोड देखें](https://codepen.io/gaearon/pen/EmmGEa?editors=0010)**
 
-For each move in the tic-tac-toes's game's history, we create a list item `<li>` which contains a button `<button>`. The button has a `onClick` handler which calls a method called `this.jumpTo()`. We haven't implemented the `jumpTo()` method yet. For now, we should see a list of the moves that have occurred in the game and a warning in the developer tools console that says:
+टिक-टैक्-टो के खेल के इतिहास में प्रत्येक चाल के लिए, हम एक सूची आइटम `<li>` बनाते हैं जिसमें एक बटन होता है `<button>`। बटन में एक  `onClick` हैंडलर होता है, जो कि `this.jumpTo()` नामक एक मेथड कॉल करता है। 
+हमने अभी तक `jumpTo()` मेथड बनाई नहीं की है। अभी के लिए, हमें खेल में होने वाली चालों की सूची दिखनी चाहिए और डेवलपर टूल्स कंसोल में एक चेतावनी दी गई है जो कहती है:
 
->  Warning:
->  Each child in an array or iterator should have a unique "key" prop. Check the render method of "Game".
+>  चेतावनी:
+>  एक array या पुनरावृत्त में प्रत्येक चाइल्ड के पास एक अद्वितीय "कुंजी" प्रोप होना चाहिए। "Game" की रेंडर मेथड की जाँच करें।
 
-Let's discuss what the above warning means.
+आइए चर्चा करें कि उपरोक्त चेतावनी का क्या मतलब है।
 
-### Picking a Key {#picking-a-key}
+### एक कुंजी उठाना {#picking-a-key}
 
-When we render a list, React stores some information about each rendered list item. When we update a list, React needs to determine what has changed. We could have added, removed, re-arranged, or updated the list's items.
+जब हम एक सूची सौंपते हैं, तो React प्रत्येक रेंडर सूची आइटम के बारे में कुछ जानकारी संग्रहीत करती है। जब हम किसी सूची को अपडेट करते हैं, तो React को यह निर्धारित करने की आवश्यकता होती है कि क्या बदल गया है। हम सूची के आइटम को जोड़, हटा, पुन: व्यवस्थित या अपडेट कर सकते थे।
 
-Imagine transitioning from
+से संक्रमण की कल्पना करो
```

Comment:
unclear translation

Reply:
`इस बदलाव के बारे में इमेजिन करें`
?

Reply:
sochiye agar hum is sthiti is tarah badalte hain ki hum <code> se <code> main jaenge.

Reply:
> sochiye agar hum is sthiti is tarah badalte hain ki hum `` se ` main jaenge` ``

This sounds a bit odd, or I am not getting it correctly. How about this?

<code>
सोचिये अगर हम state इस तरह बदलते हैं की यह लिस्ट

```html
<li>Alexa: 7 tasks left</li>
<li>Ben: 5 tasks left</li>
```

इसमें बदल जाएगी

```html
<li>Ben: 9 tasks left</li>
<li>Claudia: 8 tasks left</li>
<li>Alexa: 5 tasks left</li>
```
</code>

Reply:
cool